src/tree.js

```diff
@@ -83,16 +84,23 @@ export function createInternal(vnode, parentInternal) {
 		props,
 		key,
 		ref,
+		data: null,
+		rerender: null,
+		flags,
 		_children: null,
 		_parent: parentInternal,
 		_vnodeId: vnodeId,
 		_dom: null,
 		_component: null,
-		flags,
 		_context: null,
 		_depth: parentInternal ? parentInternal._depth + 1 : 0
 	};
 
+	if (typeof type == 'function') {
+		internal.rerender = enqueueRender.bind(null, internal);
```

Comment:
I think we could get a bit of a performance boost by skipping this bind:

```js
internal.render = enqueueRender;

// then to enqueue a render for an internal:
x._internal.render(x._internal);
```

(using `render` here should compress better than `rerender`)